libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -247,27 +253,27 @@ class MapboxTripSession(
     }
 
     private fun updateRawLocation(rawLocation: Location) {
+        locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
         ioJobController.scope.launch {
-            rawLocation.let {
-                navigator.updateLocation(it)
-            }
+            navigator.updateLocation(rawLocation)
+            lastLocationUpdateTimeMillis = Date().time
+            statusUpdateTimer.stopJobs()
         }
-        locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
-        if (this.rawLocation == null) {
-            fireOffStatusPolling()
+        fireOffStatusPolling()
+        statusUpdateTimer.startTimer {
+            if (Date().time - lastLocationUpdateTimeMillis >= STATUS_POLLING_INTERVAL) {
+                fireOffStatusPolling()
+            }
         }
```

Comment:
We update `lastLocationUpdateTimeMillis` and stop the timer from a worker thread, while we compare the value `lastLocationUpdateTimeMillis` and start the timer back up form the main thread.

This sounds like could cause various concurrency issues.